enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,300 @@
+---
+title: installer-telemetry-metrics
+authors:
+  - "@patrickdillon"
+  - "@rna-afk"
+reviewers:
+  - "@abhinavdahiya"
+  - "@brancz"
+  - "@crawford"
+  - "@sdodson"
+approvers:
+  - "@abhinavdahiya"
+  - "@brancz"
+  - "@crawford"
+  - "@sdodson"
+creation-date: 2020-02-07
+last-updated: 2020-02-07
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Installer Telemetry Metrics
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+ > 1. This requires creating an [aggregation pushgateway][weaveworks-aggregation-pushgateway] where the installer would push metrics. How would this be handled and by which team?
+ > 2. If #1 is possible, how would client authentication be administered to prevent [malicious pushing of metrics](risks-and-mitigations)? Authentication based on what? A pull secret? 
```

Comment:
1) yes I would like to see some basic forms of preventing malicious attacks (rate limits are probably sufficient)
2) pull secret auth is what we do today for openshift telemetry

Reply:
Pull secret auth sounds reasonable but let's note a couple of minor limitations:

1. We won't be able to capture failed installs where the pull-secret is the problem.
2. Commands like gather and destroy don't necessarily depend on the pull secret
I think we could live/work with these limitations.

Reply:
makes sense, I doubt we want to accept any unauthenticated telemetry though

Reply:
@brancz do you know of any example code that authenticates with pull secret to push to Prometheus?